patches/server/0994-Add-Mob-Spawner-support-to-AffectsSpawning.patch

```diff
@@ -0,0 +1,111 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tristan Krstevski <tristankrst@gmail.com>
+Date: Sat, 5 Aug 2023 19:29:57 +1000
+Subject: [PATCH] Add Mob Spawner support to AffectsSpawning
+
+
+diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
+index 18aac3da3c88f33b1a71a5920a8daa27e9723913..ceacb821d11290e6f0d8a624c910666e8a055908 100644
+--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
++++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
+@@ -565,6 +565,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
+ 
+     // Paper start - optimise checkDespawn
+     public final List<ServerPlayer> playersAffectingSpawning = new java.util.ArrayList<>();
++    public final List<ServerPlayer> playersAffectingNaturalSpawning = new java.util.ArrayList<>();
+     // Paper end - optimise checkDespawn
+     // Paper start - optimise get nearest players for entity AI
+     @Override
+diff --git a/src/main/java/net/minecraft/world/entity/EntitySelector.java b/src/main/java/net/minecraft/world/entity/EntitySelector.java
+index 3ff999734d14e2b6e7828e117f5ee32a60c26bc1..430d61bd36197879b13cf592eac9e9c3b3d3025f 100644
+--- a/src/main/java/net/minecraft/world/entity/EntitySelector.java
++++ b/src/main/java/net/minecraft/world/entity/EntitySelector.java
+@@ -45,8 +45,13 @@ public final class EntitySelector {
+     public static final Predicate<Entity> PLAYER_AFFECTS_SPAWNING = (entity) -> {
+         return !entity.isSpectator() && entity.isAlive() && entity instanceof Player player && player.affectsSpawning;
+     };
++
++    public static final Predicate<Entity> PLAYER_AFFECTS_MOB_SPAWNER_SPAWNING = (entity) -> {
++        return !entity.isSpectator() && entity.isAlive() && entity instanceof Player player && player.affectsSpawning;
```

Comment:
If I see this correctly you want to do this?
```suggestion
+        return !entity.isSpectator() && entity.isAlive() && entity instanceof Player player && player.affectsMobSpawnersSpawning;
```

Reply:
Kind of? I want to make it only impact Natural Spawns and not spawns from Spawners themself.